### DISCUSSION TITLE
GUI Enhancements -- Proposals for v1.90 (previously v1.85)

### DISCUSSION THREAD

**Discussion by K6OK** [w7sst/MorseRunner#302] [Ideas]
Now that Mike W7SST has indicated GUI improvements are on the table for the next version, I would like to share some suggested improvements to MRCE’s GUI design.  I put some thought into this about a year ago.  User interface design preferences are often subjective, so I’m intending these items to be a starting point for discussion. 

In these changes I’m looking to 1) better accommodate the user’s workflow and 2) make everything more obvious and intuitive to the user.

Based on our discussion I’ll compile a draft prototype of the .exe for you to try out.  Then of course additional refinements can be made following the group’s recommendations.

Here are areas that I see for potential improvement:

![image](https://github.com/w7sst/MorseRunner/assets/116767386/dd429603-f5e9-4a51-b28a-65ccbc51329c)

**1.  Make the entry fields and F-keys larger.**  To improve ergonomics, the “work zone” where the user does most of their work should be the most visually prominent part of the program.  Currently the user’s data entry is crowded into a small corner.  The entry fields, buttons and their font sizes should be the largest on the form.  N1MM follows this principle.

**2.  RIT Slider.**  A new user will have no idea what this mysterious green bar is.  It needs a visible title that describes it.  Right and left arrows should be provided to give the option to add RIT with a mouse click.  There should be a numerical indicator (+100, -50, etc.) indicating the degree of frequency offset.  Also a reset button to “clear RIT.”  

**3. QSO Rate Histogram.** It also needs a title, such as “Rate Chart (QSOs/5 min).”  The X-Axis only goes about 80% of the way.  The empty space on the right should be eliminated. This can be fixed by making the bars wider.

**4. Run Button / Mode Selection.** Suggest adopt the internationally accepted and well known standard Play-Pause-Stop buttons, followed by a Mode dropdown then duration, like this:
![image](https://github.com/w7sst/MorseRunner/assets/116767386/f9d2f3cf-a159-4f71-879c-06600c5394e3)
(the actual buttons will be larger/nicer than this)
The Mode dropdown would have Pile-Up, Single Calls, WPX Competition and HST Competition.   I’m not familiar with the competition modes; if Pause is not allowed, the symbol buttons can be modal, i.e., Pause can be grayed out if need be.

**5. Minimize Button.**  Concur with Issue #294, the Minimize button Windows control should be enabled and functional.

**6. Enlarge the Main Form.**  Currently the MRCE form border occupies 506 x 729 pixels, an arbitrary size? To accomodate items 1 to 5 above the form can be made larger.  To ensure compatibility with users with older computers then using XGA (1024 x 768) is a reasonable maximum.  That gives room to expand MRCE’s footprint, not to the full 1024x768, but somewhat larger than what we have now.

**7. Location of .ini file.**  Concur with Issue #256, the MorseRunner.ini file belongs in %APPDATA%.  

**8. Remember Screen Location.**  Normal practice is a program will remember where on the user’s screen it was last used.  MRCE does not do that, it always opens in the center.  Suggest adding an entry to the .ini file for this purpose.

**9. 64-bit Executables.**  It’s common practice now for programs to be offered in both 32- and 64-bit versions. MRCE probably should offer both too.

73 Jim K6OK

**Comment by w7sst** [comment]
This is great news that Jim is taking an active roll in collecting requirements and building a prototype for some UI enhancments to MRCE. Thank you Jim!

> __1. Make the entry fields and F-keys larger__. To improve ergonomics, the “work zone” where the user does most of their work should be the most visually prominent part of the program. Currently the user’s data entry is crowded into a small corner. The entry fields, buttons and their font sizes should be the largest on the form. N1MM follows this principle.

I agree with Jim's suggestion above. Below is a picture of MR from Steve, @n2ic, showing the larger font for the entry fields. I wish I had seen this sooner as I probably would have added it to MRCE. We will also want to lengthen these exchange fields as some contest exchanges are getting longer.

![image](https://github.com/w7sst/MorseRunner/assets/106113737/d6169c0e-9974-441f-9ff9-21507c43653b)

Adding to Jim's list, I'd like to add...
__10. Replace Check column with the correct data.__ Concur with Issue #266 (from Laurent, @f6fvy). I have heard this suggestion from other sources as well. With Jim's UI changes, we are growing the size of the application which also means we have a larger area for the report. This will give us plenty of room to include columns showing the actual sent data, or at least the corrected data on those rows where a bad copy occurred.

__11. Consider a Setup Dialog.__ We have been adding lots of new menu items to the Setup Menu along with lots of choices in the main UI panel. I wonder if we moved some of this setup to a Setup dialog if this would simplify the main application screen. I have not pursued this path because I am weak in UI design and implementation and would have to learn how to create a Setup dialog with tabs along the top to organize the setup sections. We would have to give considerable thought on how to organize a Setup dialog. I bring it up as part of the conversation.

@scotthibbs @WR7Q @tekenny @jr8ppg

**Comment by f6fvy** [comment]
Regarding the Setup Dialog, don't forget that several options can be modified "on the fly" during a practice session. Putting these settings in a different modal dialog will make them (almost) impossible to modify during a session. I personally like "one window for everything" like now.

**Comment by K6OK** [comment]
Thanks for the comments... am working on the prototype now.  Plan to have a working .exe for everyone in a couple of weeks.

On the Setup dialog agree with Laurent that the "on the fly" needs to be on the main form.  In addition could add a "Setup" menu item for items that don't change -- Call Sign, CQ Zone, ITU Zone, State/Province, First Name (for NAQP, SSN, etc) and perhaps CW Pitch.

Also concur with CHK column #266.  Simply show the correct call or exchange in red, otherwise leave it blank.
